feat: convert `BackendTLSPolicies` into service annotations (#6753)
Kong/kubernetes-ingress-controller: CHANGELOG.md

  By setting this flag, the secrets that are ingested will be limited to those having this label set to "true".
  This can reduce the memory usage in scenarios with a large number of giant secrets.
  [#6795](https://github.com/Kong/kubernetes-ingress-controller/pull/6795)

> - Added `BackendTLSPolicy` support. The user can now reference any kubernetes `Service`

Reviewer: ```suggestion
- Added `BackendTLSPolicy` support. The user can now reference any Kubernetes `Service`
```

  in the `BackendTLSPolicy` spec, and in case the service is used as a backend by

>   `HTTPRoute`s that reference a Kong Gateway as parent, such Backend TLS configuration

Reviewer: ```suggestion
  `HTTPRoute`s that reference a Kong Gateway as a parent, such as Backend TLS configuration
```

  By setting this flag, the secrets that are ingested will be limited to those having this label set to "true".
  This can reduce the memory usage in scenarios with a large number of giant secrets.
  [#6795](https://github.com/Kong/kubernetes-ingress-controller/pull/6795)
- Added `BackendTLSPolicy` support. The user can now reference any Kubernetes `Service`
  in the `BackendTLSPolicy` spec, and in case the service is used as a backend by
  `HTTPRoute`s that reference a Kong Gateway as parent, such Backend TLS configuration
  is applied to the service section of the Kong configuration.
  [#6712](https://github.com/Kong/kubernetes-ingress-controller/pull/6712)
  [#6753](https://github.com/Kong/kubernetes-ingress-controller/pull/6753)
- Added the flag `--configmap-label-selector` to set the label selector for `ConfigMap`s

>   to ingest. By setting this flag, the secrets that are ingested will be limited
>   to those having this label set to "true". This can reduce the memory usage in
>   scenarios with a large number of giant secrets. The default value is `konghq.com/configmap`.

Reviewer 2: ```suggestion
  to ingest. By setting this flag, the `ConfigMap`s that are ingested will be limited
  to those having this label set to "true". This limits the amount of resources that are kept in memory. The default value is `konghq.com/configmap`.
```
